openedx/features/learner_profile/views/learner_profile.py

```diff
@@ -42,10 +45,34 @@ def learner_profile(request, username):
         GET /account/profile
     """
     try:
-        return render_to_response(
+        context = learner_profile_context(request, username, request.user.is_staff)
+        # TODO: LEARNER-2554: 09/2017: Remove message and cookie logic when we no longer want this message
+        dismiss_message = False
+        if (context['own_profile'] and
+                BOOST_PROFILE_VISIBILITY.is_enabled() and
+                request.COOKIES.get('dismiss-profile-message', '') != 'True'):
+            PageLevelMessages.register_info_message(
+                request,
+                Text(_('Welcome to the new learner profile! When your profile visibility is set to "Full Profile", \
+                        other edX learners can now see your full name, the certificates you have earned, and when \
+                        you joined the edX community. You can also optionally add links to your social accounts from \
+                        the account settings page. If you would not like to share this information with other edX learners \
+                        switch to the "Limited Profile" option which only lets other edX learners see your \
+                        username and profile image. {dismiss_link}')).format(
+                    dismiss_link=HTML('<div><button class="btn-link dismiss-info-message">{dismiss_text}</button></div>').format(
+                        dismiss_text=Text(_("Dismiss"))
+                    )
+                )
+            )
+            dismiss_message = True
```

Comment:
I think you could just set the cookie here and not have to use a variable.

Reply:
The cookie is set on the response, not the request. I need to generate the response before I can set the cookie. It causes all these messy dependencies where I don't think I can generate the response before I register the page-level message or else the message won't be there.

Reply:
Ah, good point. Makes sense.